enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Filter/FilterDropdown/FilterDropdown.tsx

```diff
@@ -0,0 +1,113 @@
+import { useDisclosure } from "@mantine/hooks";
+import { useMemo, useState } from "react";
+import { match } from "ts-pattern";
+import { jt, t } from "ttag";
+
+import { FilterBadgeList } from "embedding-sdk/components/private/InteractiveQuestion/components/Filter/FilterBadgeList";
+import type { SDKFilterItem } from "embedding-sdk/components/private/InteractiveQuestion/components/Filter/hooks/use-filter-data";
+import { MultiStepPopover } from "embedding-sdk/components/private/util/MultiStepPopover";
+import type { UpdateQueryHookProps } from "metabase/query_builder/hooks";
+import { getFilterItems } from "metabase/querying/filters/components/FilterPanel/utils";
+import type { FilterColumnPickerProps } from "metabase/querying/filters/components/FilterPicker/FilterColumnPicker";
+import type * as Lib from "metabase-lib";
+
+import { useInteractiveQuestionContext } from "../../../context";
+import { ToolbarButton } from "../../util/ToolbarButton";
+import { FilterPicker } from "../FilterPicker/FilterPicker";
+
+type FilterProps = Pick<
+  FilterColumnPickerProps,
+  "withColumnItemIcon" | "withColumnGroupIcon" | "withCustomExpression"
+>;
+
+const FilterDropdownInner = ({
+  query,
+  withColumnItemIcon,
+}: UpdateQueryHookProps & FilterProps) => {
+  const filters = useMemo(() => getFilterItems(query), [query]);
+
+  const [step, setStep] = useState<"list" | "picker">("list");
+
+  const [opened, { close, toggle }] = useDisclosure(false, {
+    onOpen: () => {
+      setStep(filters.length === 0 ? "picker" : "list");
+    },
+  });
+
+  const [selectedFilter, setSelectedFilter] = useState<
+    SDKFilterItem | undefined
+  >();
+
+  const onSelectFilter = (item?: SDKFilterItem) => {
+    setSelectedFilter(item);
+    setStep("picker");
+  };
+
+  const label = match(filters.length)
+    .with(0, () => t`Filter`)
+    .with(1, () => t`1 filter`)
+    .otherwise(value => jt`${value} filters`);
+
+  const onRemoveItem = (item: SDKFilterItem) => {
+    item.onRemoveFilter();
+    if (filters.length === 1) {
+      close();
+    }
+  };
+
+  return (
+    <MultiStepPopover currentStep={step} opened={opened} onClose={close}>
```

Comment:
I find controlling the multi-step popover state with 2 props hard to reason about.

If you think about the state of the popover it can only be one of:
- closed
- open: stage 1
- open: stage 2
- open: stage n

This way we don't need to control these 2 states individually and try to make them sync to make sure `MultiStepPopover` works as expected.

The parent component can only focus on a single state machine that transitions the state of the multi-step popover instead of 2.

We can compute `open` prop in a number of ways.
1. provide `closeStep` name
2. `null` to denote the closed state
3. infer that from the step name that doesn't match any of the `MultiStepPopover.Step` value.

I like 1 the most since it's simple, we could add a typedoc to explain these details. But anything works.

Reply:
after a conversation we've decided to use `null` as the `closed` state.